protobuf-lite/src/main/java/io/grpc/protobuf/lite/ProtoLiteUtils.java

```diff
@@ -78,6 +78,18 @@ public static void setExtensionRegistry(ExtensionRegistryLite newRegistry) {
     globalRegistry = checkNotNull(newRegistry, "newRegistry");
   }
 
+  /**
+   * Local cache of buffers to use for parsing.  ThreadLocal used a WeakReference internally, so
```

Comment:
Actually... it uses a weak ref for the thread id, but it _lazily_ cleans up the values. So these will be retained for a while even after thread death.

> To help deal with very large and long-lived usages, the hash table entries use WeakReferences for keys. However, since reference queues are not used, stale entries are guaranteed to be removed only when the table starts running out of space.